### PR TITLE
fix(index): correct loader sourcemaps usage

### PR DESCRIPTION
**Pull request by mattlewis92**
I was wrong about the sourcemap behaviour, turns out webpack *does* actually pass the sourcemap as the second arg to loaders, this PR restores the previous behaviour (with a perf boost by not trying to parse for an inline sourcemap if there is already a source map available)

Depends on https://github.com/webpack-contrib/istanbul-instrumenter-loader/pull/66

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ matrix:
   fast_finish: true
 install:
   - ps: Install-Product node $env:nodejs_version x64
+  - cmd: npm i -g npm@latest
```
Comment (michael-ciniawsky) [inline appveyor.yml]: Is this fixing  the current falky builds for `branches` ?

File: test/utils/loader.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import path from 'path';
+import loader from '../../src/cjs';
+
+const windowsToUnixPathSeparators = str => str.replace(/\\/g, '/');
```
Comment (michael-ciniawsky) [inline test/utils/loader.js]:
```js
const nomalize = str => str.split(path.sep).join('/')
```

Comment (michael-ciniawsky) [inline test/utils/loader.js]: [path.sep](https://nodejs.org/dist/latest-v8.x/docs/api/path.html#path_path_sep) :)

File: appveyor.yml (hunk 1 above)
Comment (mattlewis92) [inline appveyor.yml]: It's to fix appveyor on node 4, which uses npm 2.x which apparently isn't compatible with babel

File: test/utils/loader.js (hunk 2 above)
Comment (mattlewis92) [inline test/utils/loader.js]: Done 👍

**Comment by d3viant0ne** [comment]
@mattlewis92 - Just landed #66 

I'll take a look at appveyor and figure out why it's not reporting.

**Comment by mattlewis92** [comment]
@d3viant0ne done!

**Comment by mattlewis92** [comment]
Thanks @d3viant0ne 😄 Gave the RC a spin in our app and its all working smoothly now. We probably want to land https://github.com/webpack-contrib/istanbul-instrumenter-loader/pull/65 before cutting 3.0.0 final, as that PR is breaking.

**Pull request merged**